Removing outdated VS references (Part 3)

guardrex opened the pull request:
Addresses #1795: This is Part 3 of the updates, so let's call it 🦈 *Oh Hell No!* 🦈 

* Removes and hardens references for VS2013 and VS2015.
* Updates a handful of links

Commit:
Initial commit

Commit:
Updates

Commit:
Updates

Commit:
Updates

dnfclas commented:
@GuardRex,
Thanks for having already signed the Contribution License Agreement. Your agreement was validated by .NET Foundation. We will now review your pull request.
_Thanks,
.NET Foundation Pull Request Bot_

mairaw commented on .openpublishing.redirection.json:
```diff
@@ -294,6 +294,10 @@
         "redirect_url": "/visualstudio/ide/visual-studio-ide"
     },
     {
+        "source_path": "docs/visual-basic/getting-started/breaking-changes-in-visual-studio-2015.md", 
```
not so sure about this one. I think we should keep the topic but remove version from the file name and make the topic cumulative over time. I'll redirect all the C# and VB topics from VS.140 (Vs 2015) version to here soon.

GuardRex replied:
I thought this was like a landing doc for all of the versions: https://msdn.microsoft.com/en-us/library/cc714070.aspx

Will that continue to version forward with future releases (i.e., will it continue to be cumulative over time)?

GuardRex replied:
@mairaw Since we removed the 2013 version in the csharp docs and the 2015 version is currently in the VB area (and has VB is in the title), I'll await your instructions on where you want this.

mairaw commented on docs/core/docker/visual-studio-tools-for-docker.md:
```diff
@@ -19,8 +19,8 @@ Developing and debugging your application in a Docker container can be a ceremon
 ## Prerequisites
-- [Microsoft Visual Studio 2015 Update 3](https://www.visualstudio.com/downloads/download-visual-studio-vs)
-- [.NET Core 1.0.1 - VS 2015 Tooling Preview 2](https://go.microsoft.com/fwlink/?LinkID=827546)
+- [Microsoft Visual Studio 2017](https://www.visualstudio.com/)
```
@spboyer @BillWagner wondering if this topic needs to be rewritten further. Docker support is now enabled as part of the workloads and not as a separate extension, correct?

BillWagner replied:
Yes, Docker tools are installed with Visual Studio 2017, if you install the ASP.NET Workload.

mairaw replied:
So @GuardRex, these instructions need to be reviewed. You no longer download the tool, you get it with the workloads (I believe it comes with the .NET Core workload too, need to check)

mairaw reviewed (commented):
Thanks @GuardRex. I need to stop the review now so just sending the comments I have up until this point. Some of the .NET Core articles should not be part of this sweep because they're larger updates for the csproj project. I'll comment on that later.

rpetrusha commented on docs/csharp/language-reference/language-specification.md:
```diff
@@ -33,8 +33,6 @@ translation.priority.ht:
-  
-You can download version 5.0 of this specification from the [Microsoft Download Center](http://www.microsoft.com/download/details.aspx?id=7029). If you've installed Visual Studio 2015, you can also find the specification on your computer in the Program Files (x86)/Microsoft Visual Studio 14.0/VC#/Specifications/1033 folder. If you have another version of Visual Studio installed or if you installe ...
 
 Version 6.0 of the specification is not currently available, but an unofficial draft exists in [the dotnet/csharplang GitHub repository](https://github.com/dotnet/csharplang/blob/master/spec/README.md).
```
In this case, I think that we want to provide the link tot he unofficial draft, but we also want to provide a link/information about the last official version of the C# specification.

guardrex committed:
Revert removal of last official release spec

GuardRex commented:
@rpetrusha I reverted the removal of that section and fixed a merge conflict.

GuardRex commented:
@rpetrusha The build finished. I think you checked it earlier (?) ... All good to review+merge? 🎸

GuardRex commented:
Stand-by while I take another look at this one.

Commit:
Breaking changes work #1

BillWagner commented on docs/csharp/programming-guide/interop/walkthrough-office-programming.md:
```diff
@@ -43,7 +43,11 @@ Both languages enable embedding of type information, which allows deployment of
+=======
+ You must have Microsoft Office Excel 2007 or later and Microsoft Office Word 2007 or later installed on your computer to complete this walkthrough.  
+>>>>>>> Updates
   
```
This merge conflict needs to be resolved.

GuardRex replied:
@BillWagner I'll take care of that.

Also, @mairaw floated the idea of keeping the breaking changes topic.

> I think we should keep the topic but remove version from the file name and make the topic cumulative over time. I'll redirect all the C# and VB topics from VS.140 (Vs 2015) version to here soon.

Since we removed the 2013 version in the csharp docs and the 2015 version is currently in the VB area (and has VB is in the title), I need guidance on what should be done with the breaking changes topic.

mairaw replied:
The 2013 topic will still exist in msdn. If we remove the 2015 topic, it's gone forever, So unless we don't want to have the topic, we should keep it.

GuardRex replied:
@mairaw This PR thus far keeps it, but it renames it from ...
```
docs/visual-basic/getting-started/breaking-changes-in-visual-studio-2015.md
```
to
```
docs/visual-basic/getting-started/breaking-changes-in-visual-studio.md
```
What we'd be missing is a similar doc ...
```
docs/csharp/getting-started/breaking-changes-in-visual-studio.md
```
... over in the csharp area.

At least today, this doesn't seem to provide any useful content. There are no "breaking changes" to show AFAICT from its current content. Should they be kept around in case in the future there are language-related VS breaking changes to report?

Four options:
1. Keep the doc but also keep the 2015 name on it
2. Keep the doc and rename it as done in the PR to remove the year
3. Keep the doc (either 1 or 2 above) and mirror it over in the csharp area
4. Kill this doc off :gun:

mairaw replied:
if the topic is not useful, just kill it!

Commit:
Fix merge conflict

Commit:
Updates

GuardRex commented:
@BillWagner @mairaw We have a breaking changes doc in the VB Getting Started area. It doesn't do much more than point the reader to the VS 2017 docs. We had a similar one for VS 2013 in the csharp area that we removed. It had similar content to this one.

How do you want to proceed with this one? Would you like me to reconstitute the one we removed in the csharp area?

mairaw commented on docs/core/porting/libraries.md:
```diff
@@ -13,23 +13,23 @@ ms.assetid: a0fd860d-d6b6-4659-b325-8a6e6f5fa4a1
 # Porting to .NET Core - Libraries
 
-With the release of .NET Core 1.0, there is an opportunity to port existing library code so that it can run cross-platform.  This article discusses the .NET Standard Library, unavailable technologies, how to account for the smaller number of APIs available on .NET Core 1.0, how to use the tooling that ships with .NET Core SDK Preview 2, and recommended approaches to porting your code.
+With the release of .NET Core 1.0, there is an opportunity to port existing library code so that it can run cross-platform. This article discusses the .NET Standard Library, unavailable technologies, how to account for the smaller number of APIs available on .NET Core 1.0, how to use the tooling that ships with .NET Core SDK Preview 2, and recommended approaches to porting your code.
```
this topic needed a more extensive revision. Just updating the version was not enough here but @cartermp already has a PR here: https://github.com/dotnet/docs/pull/1792. But since you've cleaned the topic a bit of extra spaces, I'm willing to leave this one here.

mairaw commented on docs/core/tools/using-ci-with-cli.md:
```diff
@@ -78,12 +78,12 @@ for more information.
 ### AppVeyor
 
-The [appveyor.com ci](https://www.appveyor.com/) has .NET Core SDK preview2 already installed in the build worker image `Visual Studio 2015`.
+The [appveyor.com ci](https://www.appveyor.com/) has .NET Core SDK 1.0.1 already installed in the build worker image `Visual Studio 2017`.
```
this one is another one that needs a more extensive revision and is covered here: https://github.com/dotnet/docs/pull/1556

mairaw reviewed (commented):
Left a few comments.

Commit:
React to feedback

GuardRex commented:
@mairaw I took a stab 🔪 at the Docker language. Not being a Docker pro, I'd like feedback there. I think ur saying on the other two to let the revisions go in on those other two PR's to get them fixed up; I didn't make changes for those here. It's ready for another look. :eyes:

BillWagner commented on docs/core/docker/visual-studio-tools-for-docker.md:
```diff
@@ -1,32 +1,31 @@
 
-> [!NOTE]
->The current version targets Linux Docker containers, with Windows Containers coming soon.
+[Microsoft Visual Studio 2017](https://www.visualstudio.com/) supports building and debugging .NET Framework web/console applications using Windows containers or .NET Core web/console applications using Linux containers. The [Docker for Windows](https://docs.docker.com/docker-for-windows/install/) install package includes everything you need to run Docker on a Windows system.
```
We now support Windows containers, in addition to Linux containers.

The docker tools are installed as part of the .NET Core workload with the Visual Studio 2017 installer. The link isn't needed.

BillWagner reviewed (changes requested):
@GuardRex I reviewed the Docker changes and had one change.

/cc @mairaw

Commit:
React to feedback

GuardRex commented:
@BillWagner Revised. I suppose since this topic explicitly pertains to .NET Core apps (the instructions have the reader create such an app), we don't need to mention other workloads that may include the Docker support when installing VS. If not, I'll need to find out which other workloads install Docker support and mention them.

BillWagner commented:
@GuardRex The only workload I saw Docker tools listed in was .NET Core. I did look before I commented.

GuardRex commented:
Ah ... fair enough. Cool.

GuardRex commented:
Looks like they should update this: https://github.com/Microsoft/DockerTools ... I'll note it to them.

GuardRex commented:
@BillWagner See Devin's remark at https://github.com/Microsoft/DockerTools/issues/10 ... Did I misunderstand your instruction in terms of the link to *Docker for Windows* that I removed? ... https://docs.docker.com/docker-for-windows/install

mairaw commented:
Yep, our comments were about the VS tooling that is no longer a separate download.

GuardRex commented:
This was the one that threw me a bit: https://github.com/dotnet/docs/pull/1811#discussion_r113813075

No probs. I'll revert that back.

Commit:
Revert Docker for Windows removal

GuardRex commented:
Let's see if this language works. Since VS prob can't debug without the runtime, I combine them to say what the reader can do with *both* installed.

Commit:
Add note to install .NET Core workload with VS

BillWagner commented:
I just read the Docker install notes, and that's accurate.

mairaw commented:
Ok to merge this now @BillWagner?
